content/en/04-tutorials/03-integrations/02-grafana.md

```diff
@@ -45,186 +45,215 @@ Before you can create a dashboard, you will need to add Pixie as a datasource:
+2. The description of the script in the **Script** drop-down menu tells you which Grafana [visualization panel](https://grafana.com/docs/grafana/latest/visualizations/) type to use with the query. Select the **Time series** visualization type from the top right drop-down menu.
 
-2. Copy & paste this query into the Query Editor.
+3. Press the `Run Script` button.
 
-```python
-# Import Pixie's module for querying data.
-import px
+> This script creates a time series graph of HTTP throughput per service. You should see data plotted over time, but the exact numbers will vary depending on the traffic in your cluster.
+
+::: div image-xl
+<svg title='Output of the "HTTP Request Throughput by Service" Pixie Grafana plugin script.' src='grafana/service-throughput.png'/>
+:::
 
-# Load data from Pixie's `http_events` table into a Dataframe.
-df = px.DataFrame(table='http_events', start_time=__time_from)
+### Add column filtering / grouping
 
-# Add K8s metadata context.
-df.service = df.ctx['service']
-df.namespace = df.ctx['namespace']
+You can apply filtering / grouping to the output columns of the pre-written scripts provided by the Pixie Datasource Plugin. Let's look at a script that creates a table of raw HTTP requests flowing through your cluster.
 
-# Bin the 'time_' column using the interval provided by Grafana.
-df.timestamp = px.bin(df.time_, __interval)
+1. From the **Query** editor tab, select the `Raw HTTP Events (Long Format)` option from the **Script** drop-down selector.
 
-# Group data by unique pairings of 'timestamp' and 'service'
-# and count the total number of requests per unique pairing.
-per_ns_df = df.groupby(['timestamp', 'service']).agg(
-        throughput_total=('latency', px.count)
-    )
+2. The description of the script in the **Script** drop-down menu tells you that this script should be used with the Table visualization panel. Select the Table panel type from the top right drop-down menu.
 
-# Calculate throughput by dividing # of requests by the time interval.
-per_ns_df.request_throughput = per_ns_df.throughput_total / __interval
-per_ns_df.request_throughput = per_ns_df.request_throughput * 1e9
+3. Press the `Run Script` button.
 
-# Rename 'timestamp' column to 'time_'. The Grafana plugin expects a 'time_'
-# column to display data in a Graph or Time series.
-per_ns_df.time_ = per_ns_df.timestamp
+> This script shows all of the HTTP requests flowing through your cluster.
 
-# Output select columns of the DataFrame.
-px.display(per_ns_df['time_', 'service', 'request_throughput'])
-```
+::: div image-xl
+<svg title='Output of the "Raw HTTP Events (Long Format)" Pixie Grafana plugin script.' src='grafana/http-events.png'/>
+:::
 
-This PxL query outputs a table of timeseries data for HTTP request throughput per service
+4. You can add or remove columns from the output table using the `Columns Display` selector. After making your selection, press `Run Script` to update the panel.
```

Comment:
I think it actually automatically runs the script for you as you make changes, so you don't even need to press "run script".